app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -30,6 +33,44 @@ class WallpaperManager(private val settings: Settings) {
             wallpaperContainer.setBackgroundResource(newWallpaper.drawable)
         }
         currentWallpaper = newWallpaper
+
+        adjustTheme(wallpaperContainer.context)
+    }
+
+    private fun adjustTheme(context: Context) {
+        val mode = if (currentWallpaper != Wallpaper.NONE) {
+            if (currentWallpaper.isDark) {
+                setDarkThemePreference()
+                AppCompatDelegate.MODE_NIGHT_YES
+            } else {
+                setLightThemePreference()
+                AppCompatDelegate.MODE_NIGHT_NO
+            }
+        } else {
+            setFollowDeviceThemePreference()
+            AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
+        }
+
+        AppCompatDelegate.setDefaultNightMode(mode)
+        context.asActivity()?.recreate()
+    }
+
+    private fun setDarkThemePreference() {
+        settings.shouldUseDarkTheme = true
+        settings.shouldUseLightTheme = false
+        settings.shouldFollowDeviceTheme = false
+    }
+
+    private fun setLightThemePreference() {
+        settings.shouldUseLightTheme = true
+        settings.shouldUseDarkTheme = false
+        settings.shouldFollowDeviceTheme = false
+    }
+
+    private fun setFollowDeviceThemePreference() {
+        settings.shouldUseLightTheme = true
+        settings.shouldUseDarkTheme = false
+        settings.shouldFollowDeviceTheme = false
```

Comment:
Probably better to refactor these methods into 1 and just take the shouldUseLightTheme, shouldUseDarkTheme an d shouldFollowDeviceTheme as parameters

Reply:
Also, provide everything with a default value to `false` to simplify further.

Reply:
👍🏽